Change order status from "Підтверджено" to "На маршруті" automatically at 00:00 on the day of order fulfillment

Markiro1 opened the pull request:
dev
## Border

* [Main border ticket](https://github.com/ita-social-projects/GreenCity/issues/5034)


## Code reviewers

- [ ] @ABbondar 
- [ ] @juseti 
- [ ] @JJJazl 
- [ ] @NKorzhik 
- [ ] @LiliaMokhnatska 

## Summary of issue

The order status from "**Підтверджено**" to "**На маршруті**" is not changed and is not set automatically at **00:00** on the day of order fulfillment

1) add new method  in OrderRepository to update order status.

2) implement method in UBSManagementServiceImpl

3) add test for method in UBSManagementServiceImplTest

4) creat OrderStatusScheduler with a method to automatically update the order status at 00:00 at night, where the order status = "CONFIRMED" and date of export = current date

5) add test for OrderStatusScheduler

## Summary of change

1) added new method "updateOrderStatusOnTheDayOfExport()" in OrderRepository to update order status.

2) added method "updateOrderStatusOnTheDayOfExport()" in UBSManagementService, because checkstyle test did not pass.

3) method implementation in UBSManagementServiceImpl

4) added test for "updateOrderStatusOnTheDayOfExport" method in UBSManagementServiceImplTest

5) created OrderStatusScheduler with a method to automatically update the order status at 00:00 at night, where the order status = "CONFIRMED" and date of export = current date

6) added test for OrderStatusScheduler

## Testing approach

Has been tested on local database several times, for several time periods

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

Markiro1 committed:
1) added new method "updateOrderStatusOnTheDayOfExport()" in OrderRepository to update order status.

2) added method "updateOrderStatusOnTheDayOfExport()" in UBSManagementService, because checkstyle test did not pass.

3) method implementation in UBSManagementServiceImpl

4) added test for "updateOrderStatusOnTheDayOfExport" method in UBSManagementServiceImplTest

5) created OrderStatusScheduler with a method to automatically update the order status at 00:00 at night, where the order status = "CONFIRMED" and date of export = current date

6) added test for OrderStatusScheduler

Markiro1 committed:
1) formatted code

Markiro1 committed:
1) remove "public" modifier in OrderStatusSchedulerTest because "Code smell"

Markiro1 committed:
1) added log.info()

RomaMocherniuk commented on dao/src/main/java/greencity/repository/OrderRepository.java:
```diff
@@ -221,4 +221,20 @@ void changeReceivingStationForAllOrders(@Param("receiving_station") Long station
     void updateCancelingReason(Long orderId, String cancellationReason);
+
+    /**
+     * Method sets order status by order's status and order's date of export.
```
Javadoc does not describe what this method does

RomaMocherniuk commented on dao/src/main/java/greencity/repository/OrderRepository.java:
```diff
@@ -221,4 +221,20 @@ void changeReceivingStationForAllOrders(@Param("receiving_station") Long station
+            + "SET order_status = :order_status "
+            + "WHERE order_status = 'CONFIRMED' "
+            + "AND date_of_export = :currentDate")
+    void updateOrderStatusOnTheDayOfExport(@Param("order_status") String orderStatus,
```
Please rename the method according to the functionality it does

RomaMocherniuk commented on service/src/main/java/greencity/config/OrderStatusScheduler.java:
```diff
@@ -0,0 +1,27 @@
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+
+@Slf4j
```
never used

RomaMocherniuk commented on service/src/main/java/greencity/config/OrderStatusScheduler.java:
```diff
@@ -0,0 +1,27 @@
+    UBSManagementServiceImpl ubsManagementService;
+
+    /**
+     * System checks BD at 00.00 daily and sends a message if the status of the
```
I do not see the logic for sending messages

RomaMocherniuk commented on service/src/test/java/greencity/config/OrderStatusSchedulerTest.java:
```diff
@@ -0,0 +1,26 @@
+class OrderStatusSchedulerTest {
+
+    @InjectMocks
+    OrderStatusScheduler orderStatusScheduler;
```
Please follow the encapsulation principle

RomaMocherniuk commented on service/src/main/java/greencity/config/OrderStatusScheduler.java:
```diff
@@ -0,0 +1,28 @@
+     */
+    @Scheduled(cron = "0 0 0 * * *", zone = "Europe/Kiev")
+    public void autoChangeOrderStatusOnTheDayOfExport() {
+        log.info("Changing order status where status = \"CONFIRMED\" and date of export = day of order fulfillment");
```
please add a test for logging
https://www.baeldung.com/junit-asserting-logs

Markiro1 committed:
1) changed javadocs
2) rename "autoChangeOrderStatusOnTheDayOfExport" to autoUpdateOrderStatus
2) rename "updateOrderStatusOnTheDayOfExport()" to "updateOrderStatusToExpected()"
3) added 1 additional param in "updateOrderStatusToExpected()" method
